src/components/CommunityDetection/RunCommunityDetectionFormDialog.tsx

```diff
@@ -0,0 +1,339 @@
+import InfoIcon from '@mui/icons-material/Info'
+import {
+  Box,
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Divider,
+  FormControlLabel,
+  InputAdornment,
+  InputLabel,
+  Select,
+  SelectChangeEvent,
+  TextField,
+  Tooltip
+} from '@mui/material'
+import * as React from 'react'
+import create from 'zustand'
+import { AlgorithmEntry } from './CommunityDetectionAlgorithmModel'
+
+interface RunCommunityDetectionFormDialogProps {
+  open: boolean
+}
+
+const useRunCommunityDetectionFormDialogStore =
+  create<RunCommunityDetectionFormDialogProps>((set) => ({
+    open: false,
+  }))
+
+export const runCommunityDetectionFormDialog = (open: boolean): void => {
+  useRunCommunityDetectionFormDialogStore.setState({
+    open,
+  })
+}
+
+const formatKey = (algorithm: string, key: string): string => {
+  return `cd.${algorithm}.${key}`
+}
+
+export const RunCommunityDetectionFormDialog: React.FC = () => {
+  const { open } = useRunCommunityDetectionFormDialogStore()
+
```

Comment:
Get the current network id and the current network associated with that id:
```js
  const currentNetworkId: IdType = useWorkspaceStore(
    (state) => state.workspace.currentNetworkId,
  )
  const networks: Map<IdType, Network> = useNetworkStore(
    (state) => state.networks,
  )
```

Reply:
Refer to `Edge.ts`, `Node.ts` `GraphObject.ts` `Network.ts` for how to understand what `targetNetwork` is.  It is a simple model right now.
https://github.com/cytoscape/cytoscape-web/tree/development/src/models/NetworkModel

Here is a json example of it:

```js
{
id: 'networkId',
nodes: [{id: '0'}, {id: '1'}],
edges: [{id: 'e0', s: '0', t: '1'}]
}
```

Reply:
This is good.  I think the next step is to serialize a sif file into a Network instance.  Is there a util to do this?  Should I just read a sif file in & create a new Network?  Oh, wait, I see that there is a createFromSif function in the CyNetwork class.  Is that the right thing to use?

Reply:
It looks like the implementation of `createFromSif` is empty.  If you were planning to convert sif files to a new network I guess that would be the place to put your code.  How are you getting a sif file?  From a request?

Reply:
I was planning on using one of the sif files from the Cytoscape desktop app.  There is a folder called sampleData with 3 sif files in it (BINDhuman.sif, BINDyeast.sif, and galFiltered.sif).  When I use currently hardcoded data, "1\t2\n1\t3\n2\t3\n", I get an error back from the CD service: 

`Received non zero exit code: 1 when running algorithm for task: 7d1c24d7-6933-4dea-8854-a3d1e37b3e24`

I suspect this is due to not using real data, but am unsure.

Reply:
I don't know the answers to those questions but I will bring it up Jing and Chris and they will reach out to you.

Reply:
Hidef does not appear to like that data. Made a ticket to see what the issue is.